CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -95,6 +97,57 @@ uint8_t convert_6bit_to_8bit_colour_component(uint8_t colour_component) {
       (colour_component & mask_6bit) * static_cast<double>(0xFF) / mask_6bit));
 }
 
+double hueToRGB(double p, double q, double hue) {
```

Comment:
This method is full of magic numbers. They probably mean something to someone that knows more about color space than I do, but I would appreciate some guidance in comments or named consts. Explaining the whole thing more will be useful.